src/main/java/com/faforever/api/config/security/oauth2/JsonApiOauthMessageConverter.java

```diff
@@ -0,0 +1,39 @@
+package com.faforever.api.config.security.oauth2;
+
+import com.faforever.api.config.error.ErrorResponse;
+import com.faforever.api.config.error.ErrorResult;
+import org.springframework.http.HttpOutputMessage;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.converter.HttpMessageNotWritableException;
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
+import org.springframework.security.oauth2.common.exceptions.OAuth2Exception;
+
+import java.io.IOException;
+
+
+public class JsonApiOauthMessageConverter  extends MappingJackson2HttpMessageConverter {
+
+  @Override
+  protected void writeInternal(Object object, HttpOutputMessage outputMessage) throws IOException, HttpMessageNotWritableException {
+    super.writeInternal(transformObject(object), outputMessage);
+  }
+
+  protected Object transformObject(Object object) {
+    ErrorResponse response = new ErrorResponse();
+    if(object instanceof OAuth2Exception) {
```

Comment:
Please enable the following "Before Commit" actions in the commit dialog:
* Reformat Code
* Rearrange Code
* Optimize imports

thx :)